transactions/manager.go

```diff
@@ -0,0 +1,143 @@
+package transactions
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/centrifuge/go-centrifuge/errors"
+	"github.com/centrifuge/go-centrifuge/identity"
+	"github.com/centrifuge/go-centrifuge/protobufs/gen/go/transactions"
+	"github.com/centrifuge/go-centrifuge/utils"
+	"github.com/satori/go.uuid"
+)
+
+// Manager wraps the repository and exposes specific functions.
+type Manager interface {
+
+	// ExecuteWithinTX executes a transaction within a transaction
+	ExecuteWithinTX(ctx context.Context, accountID identity.CentID, existingTxID uuid.UUID, desc string, work func(accountID identity.CentID, txID uuid.UUID, txMan Manager, err chan<- error)) (txID uuid.UUID, done chan bool, err error)
+
+	// CreateTransaction
+	// TODO [TXManager] remove this once TX Manager update is complete
+	CreateTransaction(accountID identity.CentID, desc string) (*Transaction, error)
+	GetTransaction(accountID identity.CentID, id uuid.UUID) (*Transaction, error)
+	SaveTransaction(tx *Transaction) error
+	GetTransactionStatus(accountID identity.CentID, id uuid.UUID) (*transactionspb.TransactionStatusResponse, error)
+	WaitForTransaction(accountID identity.CentID, txID uuid.UUID) error
+}
+
+// NewManager returns a Manager implementation.
+func NewManager(repo Repository) Manager {
+	return &service{repo: repo}
+}
+
+// service implements Manager.
+// TODO [TXManager] convert this into an implementation of node.Server and start it at node start so that we can bring down transaction go routines cleanly
+type service struct {
+	repo Repository
+}
+
+// ExecuteWithinTX executes a transaction within a transaction.
+func (s *service) ExecuteWithinTX(ctx context.Context, accountID identity.CentID, existingTxID uuid.UUID, desc string, work func(accountID identity.CentID, txID uuid.UUID, txMan Manager, err chan<- error)) (txID uuid.UUID, done chan bool, err error) {
+	t, err := s.repo.Get(accountID, existingTxID)
+	if err != nil {
+		t = newTransaction(accountID, desc)
+		err := s.SaveTransaction(t)
+		if err != nil {
+			return uuid.Nil, nil, err
+		}
+	}
+	done = make(chan bool)
+	go func(ctx context.Context) {
+		err := make(chan error)
+		go work(accountID, t.ID, s, err)
```

Comment:
This call will set up the tx.Task["smtg"] status? Or will it be done as part of the below status update?